src/mcmc/particle_mcmc.jl

```diff
@@ -15,38 +15,36 @@ Sequential Monte Carlo sampler.
 
 $(TYPEDFIELDS)
 """
-struct SMC{space,R} <: ParticleInference
+struct SMC{R} <: ParticleInference
     resampler::R
+
+    # Specifying this constructor stops the creation of default constructors. We rather
+    # define our own outer constructors below.
+    SMC{R}(resampler::R) where {R} = new{R}(resampler)
 end
 
 """
-    SMC(space...)
-    SMC([resampler = AdvancedPS.ResampleWithESSThreshold(), space = ()])
-    SMC([resampler = AdvancedPS.resample_systematic, ]threshold[, space = ()])
+    SMC()
+    SMC([resampler = AdvancedPS.ResampleWithESSThreshold()])
```

Comment:
```suggestion
    SMC([resampler = AdvancedPS.ResampleWithESSThreshold()])
```

The first line is a sub-case of the second.